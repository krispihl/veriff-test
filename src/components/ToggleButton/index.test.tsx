import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { ToggleButton } from './index';

describe('ToggleButton', () => {
    it('renders ToggleButton component correctly', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: false,
        }

        const { container } = render(<ToggleButton check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Yes', { exact: true})).toBeInTheDocument();
        expect(screen.getByText('No', { exact: true})).toBeInTheDocument();
    });

    it('renders disabled ToggleButton correctly', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: true,
        }

        const { container } = render(<ToggleButton check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getAllByRole('button')[0]).toHaveClass('disabled');
        expect(screen.getAllByRole('button')[1]).toHaveClass('disabled');
    });

    it('renders ToggleButton correctly when YES is active button', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: false,
            value: 'yes'
        }

        const { container } = render(<ToggleButton check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getAllByRole('button')[0]).toHaveClass('activeButton');
        expect(screen.getAllByRole('button')[1]).not.toHaveClass('activeButton');
    });

    it('renders ToggleButton correctly when NO is active button', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: false,
            value: 'no'
        }

        const { container } = render(<ToggleButton check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getAllByRole('button')[0]).not.toHaveClass('activeButton');
        expect(screen.getAllByRole('button')[1]).toHaveClass('activeButton');
    });
})