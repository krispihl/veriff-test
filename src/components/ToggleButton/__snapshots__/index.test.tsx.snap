// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders ToggleButton component correctly 1`] = `
<div>
  <div>
    <button
      class="leftButton"
    >
      Yes
    </button>
    <button
      class="rightButton"
    >
      No
    </button>
  </div>
</div>
`;

exports[`ToggleButton renders ToggleButton correctly when NO is active button 1`] = `
<div>
  <div>
    <button
      class="leftButton"
    >
      Yes
    </button>
    <button
      class="rightButton activeButton"
    >
      No
    </button>
  </div>
</div>
`;

exports[`ToggleButton renders ToggleButton correctly when YES is active button 1`] = `
<div>
  <div>
    <button
      class="leftButton activeButton"
    >
      Yes
    </button>
    <button
      class="rightButton"
    >
      No
    </button>
  </div>
</div>
`;

exports[`ToggleButton renders disabled ToggleButton correctly 1`] = `
<div>
  <div>
    <button
      class="leftButton disabled"
    >
      Yes
    </button>
    <button
      class="rightButton disabled"
    >
      No
    </button>
  </div>
</div>
`;
